core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -3189,11 +3189,29 @@ protected Scope visitSubqueryExpression(SubqueryExpression node, Optional<Scope>
         @Override
         protected Scope visitSetOperation(SetOperation node, Optional<Scope> scope)
         {
-            checkState(node.getRelations().size() >= 2);
-
-            List<RelationType> childrenTypes = node.getRelations().stream()
-                    .map(relation -> process(relation, scope).getRelationType().withOnlyVisibleFields())
-                    .collect(toImmutableList());
+            List<Relation> relations = node.getRelations();
+            checkState(relations.size() == 2, "relation size must be 2");
```

Comment:
These stricter checks could be as a prep commit right ? It looks like adding `CORRESPONDING` options makes these check stricter